pkg/controller/gcp/compute/subnetwork.go

```diff
@@ -0,0 +1,176 @@
+/*
```

Comment:
Any thoughts on moving `subnetwork.go` and `network.go` (along with their test files) into `subnetwork` and `network` directories respectively? That would allow us to maintain the pattern of `managed.go` and `claim.go` file naming (even though in this case we may never have a claim controller for these types).

Reply:
I like this sound of this pattern.

Reply:
The kubebuilder folder structure is that you got all the kinds that belong to the same group under the same package. The directory/package separation happens in either group level or version level. See https://github.com/muvaf/kubebuilder-init/tree/master/api/v1alpha1 for same version & same group but different kind. See https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/simplified-scaffolding.md#example for same group, different version but same kind.

Reply:
For the naming, I think we can use `network.go` + `network_claim.go`

Reply:
Do we need to strictly follow the kubebuilder generated examples for controller (not API) packages? I think we're in agreement about the flatter API packages.

With regard to controller packages, I would suggest one of:

1. `pkg/controller/{cloud}/{apigroup}/{resourcekind}/{managed,claim}.go`
1. `pkg/controller/{cloud}/{apigroup}/{resourcekind}_{managed,claim}.go`

i.e.:
1. `pkg/controller/gcp/compute/subnetwork/managed.go`
1. `pkg/controller/gcp/compute/subnetwork_managed.go`

Either of these patterns would be relatively consistent with the pattern I've been attempting to establish of putting managed resource reconcilers in `managed.go` and resource claim reconcilers in `claim.go`. I think it's fine to defer reaching consistency at this point - it can wait until we're generating controllers.

Reply:
> Do we need to strictly follow the kubebuilder generated examples for controller (not API) packages? 

We don't have to. The overall flatter API packages was something that we introduced with kubebuilder v2. Given that we want flatter packaged, `pkg/controller/{cloud}/{apigroup}/{resourcekind}_{managed,claim}.go` makes sense to me.

Reply:
We can do that for all resources at once in the stacks-gcp repo.

Reply:
It sounds like you all already are aware of this, but as a potentially unnecessary reminder I would avoid making any changes to the layout or structure of the code until all this work has been merged into the GCP stack repo.  Then we can remove GCP from the main crossplane repo and make any changes to the layout in the GCP stack repo that we desire.

Reply:
@muvaf one advantage to actually separating into separate pkgs is that the controller can just be named `ManagedController{}` and `ClaimController{}` as opposed to something like `RedisManagedController{}` and `RedisClaimController{}`. This seems somewhat inconsequential right now, but when we start adding resources like `CloudsqlDatabaseNetworkRule` (or something like that), then it may be easier to add the controllers to the manager if they are uniformly named.

Reply:
> Given that we want flatter package

FWIW I don't think this is a given. My understanding is that flatter API type packages make sense for compatibility with kubebuilder tooling, but I don't know whether their tooling has strong opinions about the structure of controller code.

Reply:
> one advantage to actually separating into separate pkgs is that the controller can just be named ManagedController{} and ClaimController{} as opposed to something like RedisManagedController{} and RedisClaimController{}

@hasheddan TBH, I see this as a disadvantage. If the name of the struct is generic, call would look like `gcpredisv1alpha1.ManagedController{}` and the part before the dot is related to how the package is imported in the given go file. This can vary wildly really. That, in my opinion, makes it harder to browse the code or understand what's really meant by `ManagedController` immediately in different go files if they didn't use the exact same name for imports (which is quite hard to impose). I'd be more comfortable understanding `gcpv1alpha1.RedisController{}` and `gcpv1alpha1.RedisClaimController{}`. In stacks repo, it could just look like `v1alpha1.RedisController{}`, which is even more intuitive.

Basically no technical requirements here, both possible, but I'd go for actual resource name in the struct name even if we go ahead with nested directory structure.

> I don't know whether their tooling has strong opinions about the structure of controller code.

@negz They're going with flat structure in for controller code as well https://github.com/muvaf/kubebuilder-init/tree/master/controllers

As @jbw976 pointed out, layout change is not something we'd like to consider for this PR. I'm going ahead and resolve this conversation.